### DISCUSSION TITLE
javascript의 문제점으로 무엇이 있고, 왜 그게 문제점일까요?

### DISCUSSION THREAD

**Discussion by airman5573** [woowacourse-study/2022-woowahan-bansanghwe#118] [JavaScript]
자바스크립트의 단점은 크게 두가지라고 생각합니다.

첫째는 안정성이고

둘째는 속도입니다

자바스크립트는 컴파일과 실행이 한 프로세스로 이뤄지기 때문에 개발하는 시점에서는 오류를 잡아내기가 어렵습니다.

또한 같은 이유로 미리 컴파일해놓을 수 없고, 클라이언트가 파일을 받은 후에 컴파일 하기 때문에 실행하기 까지의 시간이 더 걸립니다. C언어처럼 미리 컴파일을 해놓았다면 클라이언트에서 컴파일 하는 과정이 생략되기 때문에 더 빠르게 실행할 수 있었을 것입니다.

**Comment by al-bur** [comment]
자기소개

**Reply by al-bur** [reply]
왜 탐구하는 개발자죠?

**Reply by al-bur** [reply]
최근에 탐구했던 것은?

**Comment by al-bur** [comment]
javascript는 왜 나왔을까요?

**Comment by al-bur** [comment]
스크립트 언어란?

**Reply by al-bur** [reply]
인터프리터 언어란 뭐가 다른거죠?

**Comment by byhhh2** [comment]
자바스크립트의 문제점은 뭐라고 생각하시나요?

**Comment by byhhh2** [comment]
왜 자바스크립트는 컴파일 언어로 안 만들었을까요?

**Comment by byhhh2** [comment]
자바스크립트 측에서 속도 증가를 위해 한 것?

**Comment by al-bur** [comment]
노드 개발할 때도 오류를 잡아내기가 어려운가요?

**Comment by byhhh2** [comment]
런타임에서도 오류를 표시해주는데 컴파일 단계가 없다는게 단점일까?

**Comment by byhhh2** [comment]
자바스크립트도 컴파일 관련 특징을 갖는가?

**Comment by byhhh2** [comment]
자바스크립트도 객체 지향일까?

**Reply by byhhh2** [reply]
객체 지향적으로 코딩한다는 말은 무슨말일까?

**Reply by byhhh2** [reply]
객체 지향의 장단점?

**Comment by byhhh2** [comment]
클래스가 자바스크립트에 미친 영향?

**Comment by al-bur** [comment]
객체 지향 프로그래밍 vs 함수형 프로그래밍

**Comment by al-bur** [comment]
최근 자바스크립트의 동향

**Comment by airman5573** [comment]
스크립트 언어 vs 인터프리터 언어 vs 컴파일 언어
